neelance/errors/errors.go

```diff
@@ -42,12 +51,23 @@ var _ error = &QueryError{}
 
 type Builder struct {
 	Errors []*QueryError
+	// ErrorMessageFn will be used to generate the error
+	// message from errors given to Error().
+	//
+	// If ErrorMessageFn is nil, err.Error() will be used.
+	ErrorMessageFn func(error) string
 }
 
 func (c *Builder) Errorf(format string, args ...interface{}) {
 	c.Errors = append(c.Errors, Errorf(format, args...))
 }
 
 func (c *Builder) Error(err error) {
-	c.Errors = append(c.Errors, Errorf("%s", err.Error()))
+	var gqlErrMessage string
+	if c.ErrorMessageFn != nil {
```

Comment:
you could eliminate this branch by setting the default error message function to `func(err error)  string { return err.Error() }`